include/dlaf/eigensolver/reduction_to_band.h

```diff
@@ -0,0 +1,67 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/util_matrix.h"
+
+#include "dlaf/eigensolver/reduction_to_band/mc.h"
+
+namespace dlaf {
+namespace eigensolver {
+
+/// Reduce a distributed lower Hermitian matrix to symmetric band-diagonal form, with `band = blocksize + 1`.
+///
+/// The reduction from a lower Hermitian matrix to the band-diagonal form is performed by an orthogonal
+/// similarity transformation Q, applied from left and right as in equation `Q**H . A . Q`, and whose
+/// result is stored in-place in @p mat_a.
+///
+/// The Q matrix is a product of elementary Householder reflectors
+/// `Q` = H(1) . H(2) . ... . H(n)
+///
+/// with `H(i) = I - tau(i) * v(i) . v(i)**H`
+///
+/// which are stored, together with the resulting band-diagonal matrix, in-place in the lower triangular
+/// part of @p mat_a.
+///
+/// In particular, @p mat_a will look like this (tile representation)
+///
+/// B ~ ~ ~ ~ ~
+/// * B ~ ~ ~ ~
+/// v * B ~ ~ ~
+/// v v * B ~ ~
+/// v v v * B ~
+/// v v v v * B
+///
+/// where each column of `v` is an elementary reflector without its first element (which is always equal
+/// to 1), `B` are the tiles containg the band-diagonal form, while `*` tiles contain both elements
+/// of the band and of the elementary reflectors.
```

Comment:
```suggestion
/// of the band (upper triangular diagonal included) and of the elementary reflectors (lower triangular diagonal excluded).
```

Reply:
I did it "offline" because of clang-format requirements.